### PR TITLE
Refactor TeraRanger One driver to support new products

### PR DESCRIPTION
**Pull request by msadowski**
At Terabee we are excited to launch a new Range Finder in the next week or so. The change to TeraRanger driver we are proposing with this pull request modifies what previously was a TeraRangerOne driver into a generic TeraRangerI2C driver, so that the code can be reused for different products.

The main change in functionality is that the I2C address is no longer hardcoded in the driver but instead is read from the `rngfnd_addr` setting.

We will appreciate any feedback you might have!

### CODE REVIEW

File: libraries/AP_RangeFinder/AP_RangeFinder_TeraRangerI2C.cpp [hunk 1]
```diff
@@ -13,43 +13,41 @@
...
 /*
    The constructor also initializes the rangefinder. Note that this
    constructor is not called until detect() returns true, so we
    already know that we should setup the rangefinder
 */
-AP_RangeFinder_trone::AP_RangeFinder_trone(uint8_t bus, RangeFinder &_ranger, uint8_t instance, RangeFinder::RangeFinder_State &_state)
+AP_RangeFinder_TeraRangerI2C::AP_RangeFinder_TeraRangerI2C(uint8_t bus, RangeFinder &_ranger, uint8_t instance, RangeFinder::RangeFinder_State &_state)
     : AP_RangeFinder_Backend(_ranger, instance, _state, MAV_DISTANCE_SENSOR_LASER)
-    , dev(hal.i2c_mgr->get_device(bus, TRONE_I2C_ADDR))
+    , dev(hal.i2c_mgr->get_device(bus, _ranger._address[_state.instance]))
```
Comment (OXINARF) [inline libraries/AP_RangeFinder/AP_RangeFinder_TeraRangerI2C.cpp]: I'd prefer this was handled the same way it is, for example, for LightWareI2C: check that address is set and pass it to the driver.

**Comment by msadowski** [comment]
@OXINARF Thanks for taking the time to look into those changes. I will update our internal wiki straight away to include information about your contribution guidelines.

I'm looking into your proposed change and I'll make our implementation consistent with LightWareI2C. 

In case of LightWare there is a defintion of `HAL_RANGEFINDER_LIGHTWARE_I2C_BUS` [used here](https://github.com/ArduPilot/ardupilot/blob/master/libraries/AP_RangeFinder/RangeFinder.cpp#L626). Should we reuse this definition? Can I make the name more generic (e.g. `HAL_RANGEFINDER_I2C_BUS`) ?

**Comment by OXINARF** [comment]
No need to reuse that macro, keep the 0/1 bus detection since that is what was already there.

**Comment by OXINARF** [comment]
Rebased, conflicts fixed, squashed, commit messages changed, merged. Thanks!

**Pull request closed without merging**